org.eclipse.xtext.xtext.ui.examples/projects/domainmodel/org.eclipse.xtext.example.domainmodel/src/org/eclipse/xtext/example/domainmodel/validation/DomainmodelValidator.xtend

```diff
@@ -50,4 +53,20 @@ class DomainmodelValidator extends AbstractDomainmodelValidator {
 		}
 	}
 
+	@Check def void checkPropertyNamesAreUnique(Entity entity) {
+		val Multimap<String, Property> name2properties = HashMultimap.create
+
+		entity.features.filter(Property).filter[name!==null && !name.isEmpty].forEach[
+			name2properties.put(name, it)
+		]
+
+		name2properties.keySet.forEach[
```

Comment:
you actually don't need to iterate over keySet and then use the key to get an entry, you coud instead directly iterate over the values

Reply:
Fixed.